app/services/search/query_builders/classified_listing.rb

```diff
@@ -78,6 +84,10 @@ def range_keys
       def range_keys_present?
         RANGE_KEYS.detect { |key| @params[key].present? }
       end
+
+      def tags_key?(term_key)
+        (term_key == TERM_KEYS[5]) && (@params[term_key].is_a? Array)
```

Comment:
In the future, I would love the ability to be able to switch between an AND query and an OR query. With that in mind, I think a better solution here might be to support both. [Here is how I handled fixing this issue for the ReadingList tag filters.](https://github.com/thepracticaldev/dev.to/pull/7425)
```
      def terms_keys
  TERM_KEYS.flat_map do |term_key, search_key|
    next unless @params.key? term_key

    values = Array.wrap(@params[term_key])

    if params[:tag_boolean_mode] == "all" && term_key == :tag_names
      values.map { |val| { term: { search_key => val } } }
    else
      { terms: { search_key => values } }
    end
  end.compact
end
```

Reply:
hey @mstruve , thanks for the suggestion. Refactored the code accordingly

Reply:
Excellent! Its getting there but we are still missing the param `tag_boolean_mode` which will tell the code whether we want to set up the filter for an AND vs OR Elasticsearch query.